R/tidyverse.R

```diff
@@ -246,3 +330,43 @@ base_and_recommended <- function() {
     "parallel", "rpart", "spatial", "splines", "stats", "stats4",
     "survival", "tcltk", "tools", "utils")
 }
+
+## returns a data frame on up to n recent GitHub releases
+## helps the user set `since` in use_tidy_thanks()
+report_releases <- function(owner = "r-lib", repo = "usethis", n = 1) {
+  ## minimalist purrr::pluck()
+  f <- function(l, what) vapply(l, `[[`, character(1), what)
+
+  res <- gh::gh(
+    "/repos/:owner/:repo/releases",
+    owner = owner, repo = repo
+  )
+  if (identical(res[[1]], "")) {
+    return(NULL)
+  }
+  res <- utils::head(res, min(n, length(res)))
+  releases <- data.frame(
+    tag_name = f(res, "tag_name"),
+    name = f(res, "name"),
+    stringsAsFactors = FALSE
+  )
+
+  get_sha <- function(ref) {
+    gh::gh(
+      "/repos/:owner/:repo/commits/:ref",
+      owner = owner, repo = repo, ref = ref,
+      .send_headers = c("Accept" = "application/vnd.github.VERSION.sha")
+    )[["message"]]
+  }
+  releases$sha <- substr(vapply(releases$tag_name, get_sha, character(1)), 1, 7)
+
+  get_when <- function(sha) {
+    gh::gh(
+      "/repos/:owner/:repo/commits/:sha",
+      owner = owner, repo = repo, sha = sha
+    )[[c("commit", "committer", "date")]]
+  }
+  releases$commit_date <- vapply(releases$sha, get_when, character(1))
```

Comment:
Is there a reason for not using `published_at`?

Reply:
Well, I stared at this myself but I think that's when you apply the tag, which we often forget to do until several days after release. I can do a little experiment to prove this.

Reply:
Oh good point. BTW I think you can skip a step because `sha` can be a branch/tag. This seems to work:

```R
gh::gh("GET /repos/:owner/:repo/commits/:sha", 
  owner = "tidyverse", 
  repo = "tidyr",
  sha = "v0.8.1"
)
```

Reply:
Yes, I confirm that `published_at` reflects the moment at which you applied the tag. Which is not what we want.